cli/src/org/partiql/cli/Cli.kt

```diff
@@ -48,35 +47,53 @@ internal class Cli(
     }
 
     override fun run() {
-        IonReaderBuilder.standard().build(input).use { reader ->
-            val inputIonValue = valueFactory.ion.iterate(reader).asSequence().map { valueFactory.newFromIonValue(it) }
-            val inputExprValue = valueFactory.newBag(inputIonValue)
-            val bindings = Bindings.buildLazyBindings<ExprValue> {
-                // If `input` is a class of `EmptyInputStream`, it means there is no input data provided by user.
-                if (input !is EmptyInputStream) { addBinding("input_data") { inputExprValue } }
-            }.delegate(globals)
+        when (inputFormat) {
+            InputFormat.ION -> runWithIonInput()
+            InputFormat.PARTIQL -> runWithPartiQLInput()
+        }
+    }
 
+    private fun runWithIonInput() {
+        IonReaderBuilder.standard().build(input).use { reader ->
+            val bindings = when (reader.next()) {
+                null -> Bindings.buildLazyBindings<ExprValue> {}.delegate(globals)
+                else -> getBindingsFromIonValue(valueFactory.newFromIonReader(reader))
+            }
+            if (reader.next() != null) {
+                val message = "As of June 2022, PartiQL requires that Ion files contain only a single Ion value for " +
+                    "processing. Please consider wrapping multiple values in a list."
+                throw IllegalStateException(message)
+            }
             val result = compilerPipeline.compile(query).eval(EvaluationSession.build { globals(bindings) })
+            outputResult(result)
+        }
+    }
+
+    private fun runWithPartiQLInput() {
+        val partiql =
```

Comment:
nit: non-descriptive variable. Should reference that we're reading in the input environment.